cylc/flow/scheduler.py

```diff
@@ -1214,62 +1213,66 @@ def _load_workflow_params(self, row_idx, row):
-            self.options.startcp = value
-            LOG.info(f"+ start point = {value}")
-        elif key in self.workflow_db_mgr.KEY_FINAL_CYCLE_POINT_COMPATS:
-            if self.is_restart and self.options.fcp == 'reload':
-                LOG.debug(f"- final point = {value} (ignored)")
-            elif self.options.fcp is None:
-                self.options.fcp = value
-                LOG.info(f"+ final point = {value}")
-        elif key == self.workflow_db_mgr.KEY_STOP_CYCLE_POINT:
-            if self.is_restart and self.options.stopcp == 'reload':
-                LOG.debug(f"- stop point = {value} (ignored)")
-            elif self.options.stopcp is None:
-                self.options.stopcp = value
-                LOG.info(f"+ stop point = {value}")
-        elif key == self.workflow_db_mgr.KEY_RUN_MODE:
-            if self.options.run_mode is None:
+        LOG.info('LOADING workflow parameters')
+        for key, value in params:
+            if not value:
+                continue
+            if key in self.workflow_db_mgr.KEY_INITIAL_CYCLE_POINT_COMPATS:
+                self.options.icp = value
+                LOG.info(f"+ initial point = {value}")
+            elif key in self.workflow_db_mgr.KEY_START_CYCLE_POINT_COMPATS:
+                self.options.startcp = value
+                LOG.info(f"+ start point = {value}")
+            elif key in self.workflow_db_mgr.KEY_FINAL_CYCLE_POINT_COMPATS:
+                if self.is_restart and self.options.fcp == 'reload':
+                    LOG.debug(f"- final point = {value} (ignored)")
+                elif self.options.fcp is None:
+                    self.options.fcp = value
+                    LOG.info(f"+ final point = {value}")
+            elif key == self.workflow_db_mgr.KEY_STOP_CYCLE_POINT:
+                if self.is_restart and self.options.stopcp == 'reload':
+                    LOG.debug(f"- stop point = {value} (ignored)")
+                elif self.options.stopcp is None:
+                    self.options.stopcp = value
+                    LOG.info(f"+ stop point = {value}")
+            elif (
+                key == self.workflow_db_mgr.KEY_RUN_MODE
+                and self.options.run_mode is None
+            ):
                 self.options.run_mode = value
                 LOG.info(f"+ run mode = {value}")
-        elif key == self.workflow_db_mgr.KEY_UUID_STR:
-            self.uuid_str = value
-            LOG.info('+ workflow UUID = %s', value)
-        elif key == self.workflow_db_mgr.KEY_PAUSED:
-            if self.options.paused_start is None:
-                self.options.paused_start = bool(value)
-                LOG.info(f'+ paused = {bool(value)}')
-        elif key == self.workflow_db_mgr.KEY_HOLD_CYCLE_POINT:
-            if self.options.holdcp is None:
+            elif key == self.workflow_db_mgr.KEY_UUID_STR:
+                self.uuid_str = value
+                LOG.info(f"+ workflow UUID = {value}")
```

Comment:
FYI, this is a negative change.

The way it was before, the string formatting only happened if the message was actually logged. With this change the string formatting happens, even if the message is not needed.

This is why the variables are provided as arguments to the logging function rather than `'+ workflow UUID = %s' % value`.

% formatting is still the documented advice.

https://docs.python.org/3/howto/logging.html#optimization

Reply:
Interesting to know 👍  However this method is only called once per start and reload so I don't think it is significant in this case.

Reply:
I'm not asking for changes, but please leave these out of future tidying.